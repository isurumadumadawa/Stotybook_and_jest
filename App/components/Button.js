import React from 'react';

import styled from 'styled-components/native';

import colors from '../config/colors';

const ButtonWrapper = styled.TouchableOpacity`
  background-color: ${colors.primary};
  padding-vertical: 14px;
  border-radius: 6px;
  border-width: 1px;
  border-color: ${colors.primary};
  margin-vertical: 7px;
`;

const ButtonWrapperOutline = styled(ButtonWrapper)`
  background-color: transparent;
  border-color: ${colors.border};
`;

const ButtonText = styled.Text`
  color: ${colors.white};
  align-self: center;
  font-size: 18px;
  font-weight: 500;
`;

const ButtonTextOutline = styled(ButtonText)`
  color: ${colors.primary};
`;

export const Button = ({
  onPress = () => {},
  children = '',
  outline = false,
}) => {
  const Button = outline ? ButtonWrapperOutline : ButtonWrapper;
  const Text = outline ? ButtonTextOutline : ButtonText;

  return (
    <Button onPress={onPress}>
      <Text>{children}</Text>
    </Button>
  );
};
