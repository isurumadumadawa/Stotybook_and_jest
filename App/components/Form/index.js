import Form from './Form';
import FeildWrapper from './FeildWrapper';
import TextInput from './TextInput';
import Switch from './Switch';

export {Form, FeildWrapper, TextInput, Switch};
