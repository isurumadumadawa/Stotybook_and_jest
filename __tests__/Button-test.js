import 'react-native';
import React from 'react';

import render from 'react-test-renderer';
import {expect} from '@jest/globals';

import {Button} from '../App/components/Button';

test('Button snapshot', () => {
  const snapShot = render.create(<Button />).toJSON();
  expect(snapShot).toMatchSnapshot();
});
