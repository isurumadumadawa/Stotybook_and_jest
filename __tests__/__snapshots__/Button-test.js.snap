// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button snapshot 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#202c41",
      "borderColor": "#202c41",
      "borderRadius": 6,
      "borderWidth": 1,
      "marginVertical": 7,
      "opacity": 1,
      "paddingVertical": 14,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "alignSelf": "center",
          "color": "#fff",
          "fontSize": 18,
          "fontWeight": "500",
        },
      ]
    }
  >
    
  </Text>
</View>
`;
